src/main/kotlin/no/nav/tilleggsstonader/sak/behandling/vent/SettPåVentService.kt

```diff
@@ -0,0 +1,164 @@
+package no.nav.tilleggsstonader.sak.behandling.vent
+
+import no.nav.tilleggsstonader.kontrakter.oppgave.Oppgave
+import no.nav.tilleggsstonader.sak.behandling.BehandlingService
+import no.nav.tilleggsstonader.sak.behandling.domain.Behandling
+import no.nav.tilleggsstonader.sak.behandling.domain.BehandlingStatus
+import no.nav.tilleggsstonader.sak.behandling.historikk.BehandlingshistorikkService
+import no.nav.tilleggsstonader.sak.behandling.historikk.domain.StegUtfall
+import no.nav.tilleggsstonader.sak.infrastruktur.exception.feilHvis
+import no.nav.tilleggsstonader.sak.infrastruktur.sikkerhet.SikkerhetContext
+import no.nav.tilleggsstonader.sak.opplysninger.oppgave.OppgaveService
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import java.time.LocalDate
+import java.util.Optional
+import java.util.UUID
+
+@Service
+class SettPåVentService(
+    private val behandlingService: BehandlingService,
+    private val behandlingshistorikkService: BehandlingshistorikkService,
+    private val oppgaveService: OppgaveService,
+    private val settPåVentRepository: SettPåVentRepository,
+) {
+
+    fun hentSettPåVent(behandlingId: UUID): StatusPåVentDto {
+        val settPåVent = settPåVentRepository.findByBehandlingIdAndAktivIsTrue(behandlingId)
+            ?: error("Finner ikke settPåVent for behandling=$behandlingId")
+        val oppgave = oppgaveService.hentOppgave(settPåVent.oppgaveId)
+
+        return StatusPåVentDto(
+            årsaker = settPåVent.årsaker,
+            kommentar = settPåVent.kommentar,
+            frist = oppgave.fristFerdigstillelse,
+            oppgaveVersjon = oppgave.versjonEllerFeil(),
+        )
+    }
+
+    @Transactional
+    fun settPåVent(behandlingId: UUID, dto: SettPåVentDto): StatusPåVentDto {
+        val behandling = behandlingService.hentBehandling(behandlingId)
+        feilHvis(behandling.status.behandlingErLåstForVidereRedigering()) {
+            "Kan ikke sette behandling på vent når status=${behandling.status}"
+        }
+        opprettHistorikkInnslag(behandling, StegUtfall.SATT_PÅ_VENT, mapOf("årsaker" to dto.årsaker))
+        behandlingService.oppdaterStatusPåBehandling(behandlingId, BehandlingStatus.SATT_PÅ_VENT)
+
+        val oppgave = hentOppgave(behandlingId)
+        // TODO må bruke enhet til saksbehandler, men 4462 er enheten til NAY. Enhetene til tiltak er uklart
+        val mappeId = oppgaveService.finnMapper("4462").single { it.navn == "10 På vent" }.id.toLong()
+        val oppdatertOppgave = Oppgave(
+            id = oppgave.id,
+            versjon = oppgave.versjon,
+            tilordnetRessurs = "",
+            fristFerdigstillelse = dto.frist,
+            beskrivelse = SettPåVentBeskrivelseUtil.settPåVent(oppgave, dto),
+            mappeId = Optional.of(mappeId),
+        )
+        val oppgaveResponse = oppgaveService.oppdaterOppgave(oppdatertOppgave)
```

Comment:
Kanskje trekke ut dette til en egen funksjon som heter noe ala `oppdaterOgFlyttOppgave`?

Reply:
[02d56d3](https://github.com/navikt/tilleggsstonader-sak/pull/227/commits/02d56d3bd3d342c34e9c5109e6ee5c0f73e31065)

Reply:
Vet ikke om den markerte riktig, men alt som hører til oppgave

Reply:
Kanskje den funksjonen kan tilpasses sånn at den fungerer for oppdater også?

Reply:
Føler det er litt forskjellige behov her for de ulike